fdbserver/DataDistributionTracker.actor.cpp

```diff
@@ -1008,6 +1009,44 @@ ACTOR Future<Void> dataDistributionTracker(Reference<InitialDataDistribution> in
 			when(Promise<int64_t> req = waitNext(getAverageShardBytes)) {
 				req.send(self.maxShardSize->get().get() / 2);
 			}
+			when(ServerTeamInfo req = waitNext(triggerSplitForStorageQueueTooLong)) {
+				TraceEvent e("DDTrackerStorageServerQueueTooLongNotified", self.distributorId);
+				e.detail("Server", req.serverId);
+				e.detail("Teams", req.teams.size());
+				int64_t maxShardWriteTraffic = 0;
+				KeyRange shardToMove;
+				ShardsAffectedByTeamFailure::Team selectedTeam;
+				for (const auto& team : req.teams) {
+					for (auto const& shard : self.shardsAffectedByTeamFailure->getShardsFor(team)) {
+						for (auto it : self.shards.intersectingRanges(shard)) {
+							if (it->value().stats->get().present()) {
+								int64_t shardWriteTraffic = it->value().stats->get().get().metrics.bytesPerKSecond;
+								if (shardWriteTraffic > maxShardWriteTraffic &&
+								    shardWriteTraffic > SERVER_KNOBS->DD_MIN_SHARD_BYTES_PER_KSEC_TO_MOVE_OUT) {
+									shardToMove = it->range();
+									selectedTeam = team;
```

Comment:
Let's say you have selected team, now you have another "team" goes here. Maybe you should compare these two teams and choose one?

Reply:
Current design is that the hot server will keep move hot shards out of the server until the server queue becomes short. The rate of moving out shards is at most 1 per minute. I am open to any scheduling design, and I am happy to discuss this further.

Reply:
Oh, I see that you use `maxShardWriteTraffic` to find the hottest team. That's good.

I have an idea of quarantine shards to separate SSes and don't throttle traffic if these SSes have long queue, which needs changes at both DD and RK. Will talk with you and @dlambrig about it.

Reply:
Yes. It would be great to cooperate DD and RK to optimize throughput.